mypy/report.py

```diff
@@ -419,7 +419,8 @@ def __init__(self, reports: Reports, output_dir: str) -> None:
     # XML doesn't like control characters, but they are sometimes
     # legal in source code (e.g. comments, string literals).
     # Tabs (#x09) are allowed in XML content.
-    control_fixer = str.maketrans(''.join(chr(i) for i in range(32) if i != 9), '?' * 31)
+    control_fixer = str.maketrans(
+        ''.join(chr(i) for i in range(32) if i != 9), '?' * 31)  # type: Final
```

Comment:
How does this silence the warning in mypyc? Does `Final` make it compute the value at compile time?

Reply:
`Final` makes it compute the value once at module load time, like expected, instead of every time the object is instantiated.

For annoying cpython internals reasons, there is (as far as we have been able to find) no good way to accurately and efficiently reproduce the interactions between instance variables and class variables (where the class variable is read if the instance variable is missing). So what we do is we treat the value in a class definition as a *default* that gets evaluated and placed in the instance when the instance is created, and we warn if it is something nontrivial.